### DISCUSSION TITLE
RFI at U=0 in MeerKAT observations

### DISCUSSION THREAD

**Discussion by Fil8** [caracal-pipeline/caracal#1398] [General]
## Stripes in the HI-datacubes
 
 Coherent artefacts such as horizontal stripes are visibile in most datacubes of MeerKAT observations. Even though, these artefacts are present in a single MeerKAT track (5 hours observations) both in the 32k-correlator mode and in the zoom-mode, at their original resolution, their presence becomes evident when binning multiple channels together. The left panels of the figures below show images from two datacubes of open-time and MHONGOOSE observations where 20 or 100 channels (respectively, 26.1 and 6.531 kHz wide) were binned together. Horizontal stripes clearly appear in the images. This effect is also visible in the MeerKAT Fornax Survey observations. In particular, in a 5-hours MFS observation, the stripes have a peak amplitude of $60\mu$Jy b$^{-1}$, while the noise is $\sim100\mu$Jy b$^{-1}$. Assuming that the stripes add coherently, then in $50$ hours the noise goes down to $30\mu$Jy b$^{-1}$ while the stripes have stayed the same, hence their amplitude is $\sim 2$ times the noise level.

Open-time observation
![J0716-29_stn_tot](https://user-images.githubusercontent.com/9263942/149158035-dc29a28d-ac11-40b1-aa80-491d7d833d5e.png)

MHONGOOSE observation
![J0419-54_r2_tot](https://user-images.githubusercontent.com/9263942/148993059-b1095fe2-8c69-4412-a205-1fb64a6ed3bd.png)

## RFI at u=0

A periodic function in the image plane corresponds to a double point function in the $(u,v)$ plane. Hence, the FFT of the datacubes affected by the stripes should show strong signal along the vertical direction. In particular, if the stripes are characterised by only one frequency, we expect to find 2 hot pixel regions in the FFT at $u=0$ and $v=\pm V$ (similarly to the open time observation). If the stripes are characterised by multiple frequencies we expect to see a vertical line of hot pixels in the FFT (similarly to the MHONGOOSE observation).  

RFI at $u=0$ is visible in most observations of all MHONGOOSE galaxies and of the Fornax Cluster but is not constant between different observations. The Figures below shows that when observing J0419-54, tracks with low elevation ($\sim 40^\circ$, top panels) have have stronger stripes than tracks with high elevations ($>70^\circ$, bottom panels).  

Low Elevation
![J0419-54_r4_tot](https://user-images.githubusercontent.com/9263942/149158097-c19def52-8f18-450b-8e95-4cfdd510f58d.png)

High Elevation
![J0419-54_s4_tot](https://user-images.githubusercontent.com/9263942/148993342-dd9c4b57-ef3e-41f3-a24a-2c77a18e8c98.png)

Observation of the same object, taken at similar azimuth and elevation do not always show the same level of stripes. This may be due because the analysis of the stripes is done on visibilities that have already been flagged by `CaraCAL`.

## Elevation/azimuth dependence

The elevation dependence of the stripes strength seems to be present also within the same observing track.

![J0419-54_r4_All](https://user-images.githubusercontent.com/9263942/149158156-2380f336-3be6-4d52-a6d1-0f6469db57f9.png)

To gain better insights on when the stripes appear, we study the presence of the stripes in all scans of all observations of the MHONGOOSE galaxies taken so far (\ie: $\sim 5$ scans per 129 observations distributed among 25 galaxies) and of two Fornax Observations. Given their heterogeneous distribution in the sky, and that they have been observed at different times throughout the last years, we investigate if the presence and strength of the stripes changes based on the azimuth and elevation of the targets of the observations (\ie\ where in the sky the \meer\ antennas are pointing). The presence of the stripe is estimated as follows: in the FFT we compute the $99.9999$ percentile of the amplitudes. When a vertical line is present, When more than $55\%$ of the points above the 99.9999 percentile are located along the u=0 coordinate, the horizontal stripes appear in the image. Hence, the left panel of the figure shows the distribution of scans with stripes, while the right panel shows scans without the stripes. When MeerKAT is pointing at high elevations and towards the North stripes are less present than when the telescope is pointing towards the South at low elevations.

![IMG_9427](https://user-images.githubusercontent.com/9263942/148994082-42754f9c-54df-45a1-a437-3296c380195b.jpeg)

This plot is still a work in progress. We want to make wedges in el/az and average the presence of the stripes to highlight better this dependency.

## Flagging strategy
For each observation we split the continuum subtracted dataset per scan. For each scan we build an image binning together 100 channels in the line free frequency range. Of this image we compute the FFT with a pixel size of 50$\lamdba$. In the FFT, we identify the pixels with amplitudes above the 99.9999 percentile of the image. The coordinates of these pixels identify where the stripe is located. Hence, we flag in the original dataset all visibilities with the coordinates of the hot pixels of the FFT. Since the stripe is a vertical feature in the FFT, to make sure to flag also pixels that may contribute to the stripes but have not been identified by the 99.9999 percentile cutoff, we double the extent of the flagging regions along the v-direction.

In the channels where HI is present, is not possible to identify the stripe in the FFT. For this reason we use the FFT 100-channels line-free image to build a mask of the stripe in the (u,v) plane. The mask is used to flag all channels of the datacubes (including the ones with line emission/absorption).

This method allows us to get rid of the stripes in the datacubes by flagging on average per scan $\sim 1\%$ of the total visibilities, and does not significantly increase the noise in the datacube, while getting rid of the horizontal artefacts. Standard flagging done by SARAO and by the automated routines of the data reduction pipeline (ie CaraCAL) typically eliminates $8-13\%$ of all target visibilities.

![J0419-54_r2_All](https://user-images.githubusercontent.com/9263942/149007729-36d7a514-7868-4a5b-a034-68b87949ea49.png)
![J0419-54_r2_All_qrtFlag](https://user-images.githubusercontent.com/9263942/149007862-4220280e-badf-41dd-bf84-0e19cf9ca643.png)

In principle, we can also try to use the sunblocker's statistics to identify the stripe, i.e. the tail of the distribution of the amplitudes. This tends to flag more visibilities though. To be investigated further.

**Comment by bennahugo** [comment]
@o-smirnov asked me for some advice regarding this - I don't think this is anything to do with the correlator in particular. If you look at the u coordinate it is only hour angle dependent (see Synthesis Imaging II eqn 2-30), thus when you are low in elevation it means you are likely far away from the EW meridian and for declinations far removed from the SCP the projected tracks will move slowest when they near the NS (u=0 position). In this case most of the baseline length should be contained inside w as you are pointing low on the horizon. This should have the net effect of minimizing fringewashing and therefore maximizing potential RFI on especially the short spacings. At least this is my hypothesis without further information.

My question, as this is strongest South at low elevations is whether this is just unwashed GSM from the Carnarvon/Williston towers? Have you looked at plotting this as a function of frequency.
